### PR TITLE
Add support for Zig (zls)

### PR DESCRIPTION
**Pull request by rbino**
This PR adds support for the [Zig programming language](https://ziglang.org/) using the [zls](https://github.com/zigtools/zls) language server.

### CODE REVIEW

**Review by kiennq — APPROVED** [review]

File: clients/lsp-zig.el [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+;;  client for zls, the Zig language server
+
+;;; Code:
+
+(require 'lsp-mode)
+
+(lsp-register-client
+ (make-lsp-client
+  :new-connection (lsp-stdio-connection "zls")
+  :major-modes '(zig-mode)
```
Comment (kiennq) [inline clients/lsp-zig.el:32]: We're in transition from `:major-mode` to better `lsp-activate-on`, so you may want to change this like as 
```suggestion
  :activation-fn (lsp-activate-on "zig")
```

Comment (kiennq) [reply clients/lsp-zig.el:32]: @yyoncho Should we mention this in the docs when creating new client?

Comment (rbino) [reply clients/lsp-zig.el]: @kiennq thanks for the heads-up, fixed it

Comment (yyoncho) [reply clients/lsp-zig.el]: @kiennq yes, we should.

**Review by yyoncho — APPROVED** [review]

**Comment by yyoncho** [comment]
Thank you for contributing to `lsp-mode`!

**Pull request merged**